app.ts

```diff
@@ -0,0 +1,113 @@
+﻿let totalRecords = 0;
+let indexPos = 0;
+
+class AjaxCall {
+	url: string = "http://localhost:2050";
+	indexStart: number;
+	indexFinish: number;
+	fullQuery: string;
+	request: string;
+	objHttpReq: any;
+
+	doAjaxCall(request: string, callback: Function, start?: number, finish?: number) {
+		this.request = request;
+		this.indexStart = start;
+		this.indexFinish = finish;
+		if (start >= 0) { this.fullQuery = this.url + "/" + request + "?from=" + start + "&to=" + finish; }
+		else { this.fullQuery = this.url + "/" + request; }
+		this.objHttpReq = new XMLHttpRequest();
+		this.objHttpReq.onreadystatechange = () => this.onRStateChange(callback);
+		this.objHttpReq.open("Get", this.fullQuery, true);
+		this.objHttpReq.send();
+	}
+
+	onRStateChange(callback: Function) {
+		if (this.objHttpReq.readyState === 4 && this.objHttpReq.status === 200) {
+			let data = JSON.parse(this.objHttpReq.responseText);
+			(callback)(data, this.indexStart, this.indexFinish);
+		}
+		// Handle error responses here if required
+	}
+}
+
+// Create table header row based on columns received
+function populateColumns(columns: JSON) {
+	let columnArray = Object.keys(columns).map(function (k) { return columns[k]; });
+	let htmlTable = document.getElementById("table-header");
+	let tr = document.createElement('tr');
+	let columnCount = columnArray.length;
+	for (let i = 0; i < columnCount; i++) {
+		let th = document.createElement('th');
+		th.innerHTML = columnArray[i];
+		tr.appendChild(th);
+	}
+	htmlTable.appendChild(tr);
+}
+
+// Total record counter update.  Only called once on page load
+function updateRecordCounter(recordCount: JSON) {
+	totalRecords = Number(JSON.stringify(recordCount));
+	document.getElementById("page-total").innerHTML = String(totalRecords);
+}
+
+// Update table with content received from API call
+function updateGrid(records: JSON, start: number, finish: number) {
+	indexPos = start;
+	document.getElementById("page-progress").innerHTML = "Displaying " + start + " to " + finish + " of ";
+	let recordArray = Object.keys(records).map(function (k) { return records[k]; });
+	let tableContent = document.getElementById("table-body");
+	// Clear old table content
+	while (tableContent.firstChild) {
```

Comment:
If you remove table-body and create it again as well as add all records internally, then finally updating the DOM once, you'd gain a slight performance increase, but the increase would only be noticeable (and true) when adding 100's of thousands of records.